common/lib/client/realtimepresence.js

```diff
@@ -259,6 +258,10 @@ var RealtimePresence = (function() {
 	};
 
 	PresenceMap.prototype.put = function(item) {
+		if(Utils.arrIn([presenceAction.ENTER, presenceAction.UPDATE], item.action)) {
+			item = Utils.copy(item);
```

Comment:
We want a clone of the object with the correct prototype, so I think this should be `PresenceMessage.fromValues()`.